smf/clickhouse/method_script.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -x
+set -o errexit
+set -o pipefail
+
+LISTEN_ADDR="$(svcprop -c -p config/listen_addr "${SMF_FMRI}")"
+LISTEN_PORT="$(svcprop -c -p config/listen_port "${SMF_FMRI}")"
+DATASTORE="$(svcprop -c -p config/store "${SMF_FMRI}")"
+DATALINK="$(svcprop -c -p config/datalink "${SMF_FMRI}")"
+GATEWAY="$(svcprop -c -p config/gateway "${SMF_FMRI}")"
```

Comment:
Same comment about checking for `unknown` here applies; i.e., https://github.com/oxidecomputer/crucible/pull/498#discussion_r1171986146

Reply:
Added similar defensive conditionals